openmdao/core/problem.py

```diff
@@ -428,7 +428,11 @@ def set_val(self, name, value, units=None, indices=None):
             Indices or slice to set to specified value.
         """
         model = self.model
-        conns = self._metadata['connections']
+        try:
+            conns = self._metadata['connections']
+        except AttributeError:
+            raise RuntimeError("Cannot call set_val before setup.")
```

Comment:
Error message should contain the name of the variable they're trying to set and should start with f"{self.msginfo}:".